service/controller/v22/resource/s3bucket/current.go

```diff
@@ -65,9 +67,12 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 				if err != nil {
 					return microerror.Mask(err)
 				}
+
+				m.Lock()
 				inputBucket.IsLoggingBucket = isLoggingBucket(bucketName, lc)
 				inputBucket.IsLoggingEnabled = isLoggingEnabled(lc)
 				currentBucketState = append(currentBucketState, inputBucket)
+				m.Unlock()
```

Comment:
That was potentially weird. Would also be good to have more unit tests for these things. Have to think a bit more about it.